HIRS_Utils/src/main/java/hirs/data/persist/certificate/attributes/V2/CertificateIdentifier.java

```diff
@@ -0,0 +1,117 @@
+package hirs.data.persist.certificate.attributes.V2;
+
+import hirs.data.persist.DeviceInfoReport;
+import java.math.BigInteger;
+import org.bouncycastle.asn1.ASN1Sequence;
+import org.bouncycastle.asn1.DERUTF8String;
+import org.bouncycastle.asn1.x509.GeneralName;
+
+/**
+ * Basic class that handles a the attribute associate with a Certificate
+ * Identifier for the component.
+ * <pre>
+ * CertificateIdentifier::= SEQUENCE {
+ *       attributeCertIdentifier [0] IMPLICIT AttributeCertificateIdentifier OPTIONAL
+ *       genericCertIdentifier   [1] IMPLICIT IssuerSerial OPTIONAL }
+ *
+ * AttributeCertificateIdentifier ::= SEQUENCE {
+ *       hashAlgorithm  AlgorithmIdentifier,
+ *       hashOverSignatureValue OCTET STRING }
+ *
+ * IssuerSerial ::= SEQUENCE {
+ *       issuer        GeneralNames,
+ *       serial        CertificateSerialNumber }
+ * </pre>
+ */
+public class CertificateIdentifier {
+
+    private static final int IDENTIFIER_NUMBER = 2;
+
+    private String hashAlgorithm;
+    private DERUTF8String hashSigValue;
+    private GeneralName issuerDN;
+    private BigInteger certificateSerialNumber;
```

Comment:
I was going to leave this empty because I wasn't sure how to expect since I didn't have a test pattern but I took a swipe in the last change to fix your other comment.